src/plugins/explore/public/components/visualizations/visualization_container.tsx

```diff
@@ -72,13 +72,25 @@ export const VisualizationContainer = () => {
 
   const visualizationRegistry = useVisualizationRegistry();
 
-  // Initialize selectedChartType and its default styles when visualizationData changes
   useEffect(() => {
     if (visualizationData && visualizationData.visualizationType) {
-      dispatch(setSelectedChartType(visualizationData.visualizationType.type));
-      dispatch(setStyleOptions(visualizationData.visualizationType.ui.style.defaults));
+      if (!selectedChartType) {
+        // default initialization on load
+        dispatch(setSelectedChartType(visualizationData.visualizationType.type));
+        dispatch(setStyleOptions(visualizationData.visualizationType.ui.style.defaults));
+        return;
+      }
+      // if URL stores a different chart type than the visualizationData; we should persist the chart type from URL
+      if (selectedChartType !== visualizationData.visualizationType.type) {
+        const chartConfig = visualizationRegistry.getVisualizationConfig(selectedChartType);
+        setVisualizationData({
+          ...visualizationData,
+          visualizationType: chartConfig as VisualizationType<ChartType>,
+        });
+        return;
+      }
```

Comment:
Hi there, IMHO, I think if we only using chart type in the URL it seems not enough. If we want to support URL sharing for the visualisation that the user manually selects, we also need fields and styleOptions. Therefore, we many not need to always synchronise type in visualization data from URL.